source/common/network/connection_impl.cc

```diff
@@ -535,6 +586,34 @@ bool ConnectionImpl::bothSidesHalfClosed() {
   return read_end_stream_ && write_end_stream_ && write_buffer_->length() == 0;
 }
 
+#ifdef __APPLE__
+bool ConnectionImpl::detectEarlyClose() {
+  auto& os_syscalls = Api::OsSysCallsSingleton::get();
+
+  if (is_uds_) {
+    ENVOY_CONN_LOG(trace, "checking for UDS early close with read disabled", *this);
+
+    int bytes;
+    socklen_t bytes_size = sizeof(int);
+    const Api::SysCallIntResult result =
+        os_syscalls.getsockopt(fd(), SOL_SOCKET, SO_NREAD, &bytes, &bytes_size);
+    ASSERT(0 == result.rc_);
```

Comment:
can you not get EAGAIN for this?

Reply:
Perhaps, but as you noted, we don't need to attempt to support this for UDS, so this call can be removed.